### DISCUSSION TITLE
How to enable refresh control within screen?

### DISCUSSION THREAD

**Discussion by ComputerFrek** [infinitered/ignite#2533] [Q&A] [answered]
Hi, how do I refreshcontrol in Screen component?
https://reactnative.dev/docs/refreshcontrol

I thought that Screen is an extension of Scroll View so I've tried this but it is not working. 
```jsx
<Screen style={$container} preset="scroll" refreshControl={<RefreshControl refreshing={loading} onRefresh={onRefresh} />}>
```

Any help would be greatly appreciated.

**Comment by frankcalise** [comment, marked as answer]
Hey @ComputerFrek , it is an extension of `ScrollView`, you can utilize the prop named `ScrollViewProps` to reach for `refreshControl`:

```jsx
<Screen
      preset="scroll"
      safeAreaEdges={["top"]}
      contentContainerStyle={$container}
      ScrollViewProps={{ refreshControl: <View /> }}
    >
```

However, `refreshing` and `onRefresh` are only props of `FlatList`.

[ScrollView refreshControl](https://reactnative.dev/docs/scrollview#refreshcontrol)
[FlatList onRefresh](https://reactnative.dev/docs/flatlist#onrefresh)
[FlatList refreshing](https://reactnative.dev/docs/flatlist#refreshing)
